src/Engine/GameObject/GameObject.cpp

```diff
@@ -10,24 +10,29 @@ GameObject::GameObject() :
     m_spatialComponent(nullptr)
 {}
 
-GameObject::~GameObject() {
-
+void GameObject::addComponent(Component & component, std::type_index typeI) {
+    m_allComponents.push_back(&component);
+    m_compsBySysT[component.systemID()].push_back(&component);
+    m_compsByCompT[typeI].push_back(&component);
+    if (typeI == std::type_index(typeid(SpatialComponent))) {
+        m_spatialComponent = dynamic_cast<SpatialComponent *>(&component);
+    }
 }
 
-/* Initialize all components */
-void GameObject::init() {
-    for (Component * c : m_allComponents) {
-        c->init();
```

Comment:
Is Component::init() called elsewhere?

Reply:
yes, by scene in the initialization queue